OnServiceChange/__tests__/handler.test.ts

```diff
@@ -0,0 +1,230 @@
+        Promise.resolve(mockApimOrganizationUserReponse)
+      )
+  }
+});
+const mockApimClient = {
+  getClient: mockGetClient
+};
+
+const mockQueryResult = {
+  command: "INSERT",
+  rowCount: 1
+} as QueryResult;
+const mockClient = {
+  connect: jest.fn().mockImplementation(() =>
+    Promise.resolve({
+      query: jest
+        .fn()
+        .mockImplementation(() => Promise.resolve(mockQueryResult))
+    })
+  )
+};
+
+const mockDocuments = [
+  {
+    subscriptionId: "01FG981SCZVVDT5E7DPZ6Z2ZR7" as NonEmptyString,
+    organizationFiscalCode: "11111111111" as OrganizationFiscalCode,
+    serviceName: "Servizi scolastici 2!" as NonEmptyString
+  },
+  {
+    subscriptionId: "01EYNQ08CFNATVH1YBN8D14Y8S" as NonEmptyString,
+    organizationFiscalCode: "01234567891" as OrganizationFiscalCode,
+    serviceName: "Lorenzo Test" as NonEmptyString
+  },
+  {
+    subscriptionI: "01EYNQ08CFNATVH1YBN8D14Y8S" as NonEmptyString,
+    organizationFiscalCode: "01234567891" as OrganizationFiscalCode,
+    serviceName: "Lorenzo Test" as NonEmptyString
+  }
+];
+
+const mockConfig = {};
+
+describe("getApimOwnerBySubscriptionId", () => {
+  it("should have valid properties", async () => {
+    const apimClient = (mockApimClient.getClient() as unknown) as ApiManagementClient;
+
+    const res = await getApimOwnerBySubscriptionId(
+      mockConfig as IDecodableConfigAPIM,
+      apimClient,
+      mockSubscriptionId
+    )();
+
+    expect(isRight(res)).toBe(true);
+    if (isRight(res)) {
+      expect(res.right).toHaveProperty("subscriptionId");
+      expect(res.right).toHaveProperty("ownerId");
+    }
+  });
+  it("should response with Left for invalid client", async () => {
+    const apimClient = mockApimClient.getClient();
+    apimClient.subscription.get.mockImplementationOnce(() =>
+      Promise.reject(
+        new Error(
+          "The provided subscription identifier is malformed or invalid."
+        )
+      )
+    );
+
+    const res = await getApimOwnerBySubscriptionId(
+      mockConfig as IDecodableConfigAPIM,
+      (apimClient as unknown) as ApiManagementClient,
+      mockSubscriptionId
+    )();
+
+    expect(isRight(res)).toBe(false);
+    if (isLeft(res)) {
+      expect(res.left).toEqual({ kind: "apimsuberror" });
+    }
+  });
+});
+
+describe("getApimUserBySubscription", () => {
+  it("should have valid properties for Organization", async () => {
+    const apimClient = (mockApimClient.getClient() as unknown) as ApiManagementClient;
+
+    const res = await getApimUserBySubscription(
+      mockConfig as IDecodableConfigAPIM,
+      apimClient,
+      mockApimSubscriptionResponse
+    )();
+    expect(isRight(res)).toBe(true);
+    if (isRight(res)) {
+      expect(res.right).toHaveProperty("id");
+      expect(res.right).toHaveProperty("firstName");
+      expect(res.right).toHaveProperty("lastName");
+      expect(res.right).toHaveProperty("email");
+      expect(res.right).toHaveProperty("kind", "organization");
+    }
```

Comment:
Since we are decoding the result and obtaining a right value, this is useless. 
If we need to test that io-ts model fits our requirements, I suggest to create a test in `models` folder

Reply:
Fix with 1c859d4